src/java.base/share/classes/sun/invoke/util/VerifyAccess.java

```diff
@@ -208,6 +208,13 @@ public static boolean isClassAccessible(Class<?> refc,
                 return true;
             }
 
+            // exports are not setup during early VM initialization
+            if (!jdk.internal.misc.VM.isModuleSystemInited()) {
```

Comment:
Should we merge this and the `lookupModule == null` early startup cases, and `assert lookupModule == refModule && refModule == Object.class.getModule()` can become `assert lookupModule == refModule && (refModule == null || refModule == Object.class.getModule())`?

Reply:
It can be simpler:

```
            // early VM startup case, java.base not defined or module system not initialized
            if (lookupModule == null || !jdk.internal.misc.VM.isModuleSystemInited()) {
                assert lookupModule == refModule;
                return true;
            }
```

Reply:
Can `Class::getModule` ever return `null` after module system is initialized? If not we can omit the `lookupModule == null` check.

Reply:
> Can Class::getModule ever return null after module system is initialized?

No.

`lookupModule == null` when module system is being initialized before java.base is defined.   So the null check needs to be there.